src/main.rs

```diff
@@ -267,17 +269,100 @@ fn build(config: &Config) {
     };
 }
 
+fn static_file_handler(dest: &str, req: Request, mut res: Response) {
+    // grab the requested path
+    let req_path = match req.uri {
+        RequestUri::AbsolutePath(p) => p,
+        _ => {
+            println!("Requset is not of the format AbsolutePath(p)");
+            std::process::exit(1);
+        },
+    };
+
+    // find the path of the file in the local system
+    // (this gets rid of the '/' in `p`, so the `join()` will not replace the
+    // path)
+    let path = PathBuf::from(dest).join(&req_path[1..]);
+
+    let serve_path = if path.is_file() {
+        // try to point the serve path to `path` if it corresponds to a file
+        path
+    } else {
+        // try to point the serve path into a "index.html" file in the requested
+        // path
+        path.join("index.html")
+    };
+
+    // if the request points to a file and it exists, read and serve it
+    if serve_path.exists() {
+        let mut file = match File::open(serve_path) {
+            Ok(f) => f,
+            Err(e) => {
+                // error!("{}", e);
+                println!("{}", e);
+                std::process::exit(1);
+            }
+        };
+
+        // buffer to store the file
+        let mut buffer: Vec<u8> = vec![];
+
+        if let Err(e) = file.read_to_end(&mut buffer) {
+            // error!("{}", e);
+            println!("{}", e);
+            std::process::exit(1);
+        }
+
+        if let Err(e) = res.send(&buffer) {
+            // error!("{}", e);
+            println!("{}", e);
+            std::process::exit(1);
+        }
+    } else {
+        // return a 404 status
+        *res.status_mut() = hyper::status::StatusCode::NotFound;
+
+        // write a simple body for the 404 page
+        let body = b"<h1> <center> 404: Page not found </center> </h1>";
+
+        if let Err(e) = res.send(body) {
+            // error!("{}", e);
+            println!("{}", e);
+            std::process::exit(1);
```

Comment:
You do this quite a lot, it's getting pretty verbose. What you could do is just return a `Result<(), String>` from `static_file_handler` and handle the error case in your closure. Then you could simply do

```rust
try!(res.send(body));
```
or

```
try!(file.read_to_end(&mut buffer));
```

which reads much nicer.